.github/workflows/ros-pkg-workflow.yml

```diff
@@ -0,0 +1,231 @@
+name: Build and pack Ros packages 
+on:
+  workflow_call:
+    inputs:
+      deploy:
+        required: false
+        type: string
+        default: 'false'
+      release:
+        required: false
+        type: string
+        default: 'false'
+
+    secrets:
+      auto_commit_user:
+        required: true
+      auto_commit_mail:
+        required: true
+      registry_user:
+        required: true
+      registry_password:
+        required: true
+      nexus_publisher_user:
+        required: true
+      nexus_publisher_password:
+        required: true
+      aws_sqs_rosdep_access_key:
+        required: true
+      aws_sqs_rosdep_secret_access_key:
+        required: true
+      github_token:
+        required: true
+      sonar_token:
+        required: true
+
+jobs:
+  Test-Ros-Component:
+    if: ${{ inputs.release == 'false' }}
+    runs-on: ubuntu-20.04
+    container:
+      image: registry.cloud.mov.ai/devops/ros-buildtools-melodic:main
+      options: --user root
+      credentials:
+        username: ${{secrets.registry_user}} 
+        password: ${{secrets.registry_password}}
+
+    steps:
+    - name: Checkout    
+      uses: actions/checkout@v2
+      with:
+        # Disabling shallow clone is recommended for improving relevancy of reporting
+        fetch-depth: 0
+        submodules: recursive
+
+    - name: Validate Repository Package.xml
+      run: |
+        expected_url="<url>$GITHUB_SERVER_URL/$GITHUB_REPOSITORY</url>"
+        VALIDATION_FILE="/tmp/verification-failed"
+        find -L ./ -type f -name 'package.xml' -exec sh -c '
+
+        search_result=$(cat $0 | grep "$1")
+
+        if [ -z "$search_result" ]:
+        then
+            echo "\e[31m$0"
+            touch $2
+        fi 
+
+        ' {} $expected_url $VALIDATION_FILE \;
```

Comment:
please simplify

Reply:
Also fix the problem of submodules. We can only validate package.xmls that are ours. Either check maintainer mail or package name containing movai OR create a new extra attribute in the export section.